2021-05-06-updating-via-direct-commands-no-longer-recommended.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: post
+title: "Updating via direct commands no longer recommended"
+categories:
+ - security
+ - announcements
+author: Andrew David Wong
+---
+
+For many years now, Qubes users have been accustomed to updating their systems using commands like `qubes-dom0-update`, `dnf update`, and `apt update` (along with several variants). Recently, however, the Qubes team has made numerous security improvements to the way updates work in Qubes OS (see below for examples). In order to benefit from these improvements, it is important that users switch from using the old direct commands to using the **Qubes Update** tool or its command-line equivalents, which are documented in our newly-revised [Updating Qubes OS](/doc/updating-qubes-os/) page.
+
+_**Note:** This change is only about **updating**. It's still fine to use direct commands for **installing** new packages._
+
+## User action summary
+
+ - **Do** keep updating your system frequently, including dom0, TemplateVMs, and StandaloneVMs (if you have any).
+ - **Avoid** direct use of old commands like `qubes-dom0-update`, `dnf update`, and `apt update` (and their variants).
+ - **Do** use the **Qubes Update** tool (or its `qubesctl` command-line equivalents), as documented in [Updating Qubes OS](/doc/updating-qubes-os/).
+
+## Security explanation and examples
+
+When you update using the **Qubes Update** tool (or its `qubesctl` command-line equivalents), you are updating via the [Salt management stack](/doc/salt/). Salt allows the Qubes developers to deliver security fixes that cannot be handled by normal package updates. For example, you can think of updating a Fedora TemplateVM with the **Qubes Update** tool as doing `dnf update` *plus* any available Salt configuration steps from the Qubes developers. In some cases, certain security bugs can only be fixed via these Salt configuration steps. They can't be handled by a package installed via `dnf` or another package manager alone.
+
+There are four main examples that illustrate the importance of updating via Salt:
+
+1. [QSB-046 / DSA-4371-1: APT update mechanism vulnerability](/news/2019/01/23/qsb-46/). Since the security flaw was in the APT update mechanism itself, the alternative patching method described in QSB-046 relied on using special Salt configuration steps to fix it. It was very important to fix APT updates *before* attempting to use APT to update normally.
+
+2. [QSB-067: Multiple RPM vulnerabilities](/news/2021/03/19/qsb-067/). Demi M. Obenour discovered several security issues in the RPM package manager. As in the previous example, these were security flaws in the update mechanism itself. In this case, Salt was used to apply a mitigation that prevented privilege escalation as well as to strengthen authentication by forcing RPM to always verify package signatures. Users who updated exclusively via `dnf` would not have received these security benefits.
```

Comment:
This should have been made available via Qubes Manager, but was not.

Reply:
I think that's a good enough reason not to recommend use of the Qube Manager for updates and to recommend only the Qubes Update tool / qubesctl. The Qube Manager update should just be made to call the Qube Update tool ASAP. It is weird and confusing that it doesn't already do this.